00_Base/src/services/sessions.service.ts

```diff
@@ -33,8 +33,8 @@ export class SessionsService {
       this.transactionRepository.getTransactionsCount(dateFrom, dateTo),
     ]);
 
-    const sessions = await this.sessionMapper.mapTransactionsToSessions(
-      transactions,
+    const sessions = this.filterBasedOnCountryCodePartyId(
+      await this.sessionMapper.mapTransactionsToSessions(transactions),
```

Comment:
Not something I'm asking u to do in this PR, but I would recommend we omit any references to the ocpp DBs from the service layer and instead define some sort of datasource layer.

Then, the service layer can be agnostic of the ocpp layer.

E.g.

```
  public async getSessions(
    fromCountryCode: string,
    fromPartyId: string,
    toCountryCode: string,
    toPartyId: string,
    dateFrom?: Date,
    dateTo?: Date,
    offset: number = DEFAULT_OFFSET,
    limit: number = DEFAULT_LIMIT,
  ): Promise<PaginatedSessionResponse> {
    
    const sessions = sessionsDatasource.getSessions(...);  
    const total = sessionsDatasource.getTotalSessions(...);  
    
    const response = buildOcpiPaginatedResponse(
      OcpiResponseStatusCode.GenericSuccessCode,
      total,
      limit,
      offset,
      sessions,
    );

    return response as PaginatedSessionResponse;
  }
```
The datasource layer would contain calls to the ocpp DBs and mappers as you have here.

This would allow us to define an Ocpp1.6 datasource and Ocpp2.0.1 datasource that we could easily swap in and out whenever we get to that point.